railties/lib/rails/generators/rails/app/templates/config/application.rb.tt

```diff
@@ -29,9 +29,11 @@ module <%= app_const_base %>
     config.load_defaults <%= build(:config_target_version) %>
 
     # Settings in config/environments/* take precedence over those specified here.
-    # Application configuration can go into files in config/initializers
+    # Most application configuration can go into files in config/initializers
     # -- all .rb files in that directory are automatically loaded after loading
-    # the framework and any gems in your application.
+    # the framework and any gems in your application. Some options however,
+    # in particular `config.time_zone` and `config.autoload_paths`, still need
+    # to be configured here.
```

Comment:
If we're going to give this guidance, I think it'd be clearer as a separate code block (with commented-out suggests for config.time_zone and .autoload_paths) that says that boot-time Rails config must go in config/application.rb.

If that doesn't meet the bar for inclusion in all newly-generated apps, I'd wager that this specialized caveat won't either. Having boot and initialization phases share config space is the problem 😅. If setting these after boot doesn't work, then perhaps we should make setting them after boot raise a RuntimeError instead.

Reply:
Think I remember seeing a commit from DHH that removed these very two configs as not common enough to call out. Can't find it now, though.

Reply:
@kaspth It's [this commit](https://github.com/rails/rails/commit/28dcadc0140dfdebe87d5e691fd709c0a9ae0bae), though I can't find the pull request for it.

@jeremy I'm not sure what to make of your comment. The text as it stands is blatantly wrong and misleading; surely you don't prefer to keep it? Anyway, your and @fxn's comments are at odds. What do you want me to change to get this merged?

Reply:
Seems like that commit from DHH was misguided.  Maybe it's true those 2 configs "are not changed nearly often enough to warrant inclusion" in application.rb, but there is a more important concern.  Inclusion prevents misconfiguration that fails silently and is difficult to troubleshoot (as reported in Issue #24748, for example).  **Proposal: This PR should restore the code DHH's commit deleted, rather than changing the guidance.**  I believe that would jive with what everyone here is saying.

(Side note: I agree the guidance is misleading in any case and should eventually be changed, as it's not safe to put application configuration in initializers - see Issue #36650.)